src/Compilers/Core/Portable/Operations/ICatchClause.cs

```diff
@@ -11,24 +9,31 @@ namespace Microsoft.CodeAnalysis.Semantics
     /// This interface is reserved for implementation by its associated APIs. We reserve the right to
     /// change it in the future.
     /// </remarks>
-    internal interface ICatchClause : IOperation // https://github.com/dotnet/roslyn/issues/22008
+    public interface ICatchClause : IOperation
     {
         /// <summary>
         /// Body of the exception handler.
         /// </summary>
         IBlockStatement Handler { get; }
+
         /// <summary>
-        /// Type of exception to be handled.
+        /// Type of the exception handled by the catch clause.
         /// </summary>
-        ITypeSymbol CaughtType { get; }
+        ITypeSymbol ExceptionType { get; }
+
         /// <summary>
-        /// Filter expression to be executed to determine whether to handle the exception.
+        /// Optional source for exception. This could be any of the following operation:
```

Comment:
>Optional source for exception. This could be any of the following operation: [](start = 12, length = 76)

It looks like the list is not complete. For example, one of the tests have a field reference expression as the ```ExceptionDeclarationOrExpression```. Consider rephrasing item 4: "Other expression for error scenarios in VB"